cypress/e2e/e2e.cy.js

```diff
@@ -0,0 +1,105 @@
+describe('Teste E2E EngageSphere', () => {
+    beforeEach(() => {
+        cy.setCookie('cookieConsent', 'accepted')
+        cy.visit('/')
+    })
+
+    it('Retorna à lista de clientes ao clicar no botão "Voltar"', () => {
+        cy.contains('button', 'View').should('be.visible').click()
+        cy.contains('button', 'Back').should('be.visible').click()
+        cy.contains('Below is our customer list.').should('be.visible')
+    })
+
+    it('Exibie o rodapé com o texto e links corretos', () => {
+        cy.contains('p', 'Copyright 2024 - Talking About Testing')
+            .should('be.visible')
+        cy.contains('a', 'Hotmart')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem')
+        cy.contains('a', 'Udemy')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://udemy.com/user/walmyr')
+        cy.contains('a', 'Blog')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://talkingabouttesting.com')
+        cy.contains('a', 'YouTube')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://youtube.com/@talkingabouttesting')
+    })
+
+    it('Exibe a saudação "Hi, there" quando nenhum nome é fornecido', () => {
+        cy.contains('h2', 'Hi there!').should('be.visible')
+    })
+
+    it('Exibe a saudação "Hi, Joe" quando o nome é fornecido', () => {
+        cy.get('#name').type('Joe')
+        cy.get('h2').should('contain', 'Hi Joe!')
+    })
+
+    it('Exibe o cabeçalho com um título, alternador de tema e um campo de entrada de texto', () => {
+        cy.contains('h1', 'EngageSphere').should('be.visible')
+        cy.get('[class^="ThemeToggle_button"]').should('be.visible')
+        cy.get('input[type="text"]').should('be.visible')
+    })
+
+    it('Abre e fecha o messenger', () => {
+        cy.get('button[class^="Messenger_openCloseButton"]').should('be.visible').click()
+        cy.get('h2').should('contain', 'How can we help you?')
```

Comment:
Veja uma alternativa mais simples.

```suggestion
        cy.contains('h2, 'How can we help you?').should('be.visible')
```

A propósito, perceba a verificação pela visibilidade do elemento.